### PR TITLE
WIP updates to node,publish, and publisher.

### PR DESCRIPTION
**Pull request by neeleym**
Simply committing some work in progress. Node, publish, and publisher were worked on here, but this will not build or run yet. A LOT of
vendor-specific leftovers in here.

### CODE REVIEW

File: rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/node.hpp [hunk 1]
```diff
@@ -28,6 +28,12 @@ create_node(
   size_t domain_id,
   const rmw_node_security_options_t * options);
 
+
+RMW_OPENDDS_SHARED_CPP_PUBLIC
+        rmw_ret_t
```
Comment (iguessthislldo) [inline rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/node.hpp]:
```suggestion
rmw_ret_t
```

File: rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/node_info_and_types.hpp [hunk 2]
```diff
@@ -33,6 +33,16 @@ get_publisher_names_and_types_by_node(
   bool no_demangle,
   rmw_names_and_types_t * topic_names_and_types);
 
+RMW_OPENDDS_SHARED_CPP_PUBLIC
+        rmw_ret_t
```
Comment (iguessthislldo) [inline rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/node_info_and_types.hpp]:
```suggestion
rmw_ret_t
```

**Comment by iguessthislldo** [comment]
Also It looks like you guys are using GNU/ACE style. Is this the style of ROS or is this a left over from Johnny working on it? If ROS doesn't have a recommended style I think using the OpenDDS style would be a good idea.

**Comment by adamsj-ros** [comment]
> Also It looks like you guys are using GNU/ACE style. Is this the style of ROS or is this a left over from Johnny working on it? If ROS doesn't have a recommended style I think using the OpenDDS style would be a good idea.

ROS offers uncrustify configs and linters to check and format the code. I feel like we can leave it as for now and run the scripts to beautify it later. To answer your question, it seems that much of ROS2 prefers this style as it matches the other RMWs.

File: rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/node.hpp (hunk 1 above)
Comment (adamsj-ros) [reply rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/node.hpp]: Style updates are helpful. We can commit but we will run a beautify and linter at some point later in the development.

Comment (iguessthislldo) [reply rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/node.hpp]: This isn't a style thing, this is a typo.

Comment (iguessthislldo) [reply rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/node.hpp]: This is just a typo.

Comment (adamsj-ros) [reply rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/node.hpp]: Missed that. Thanks.

**Comment by adamsj-ros** [comment]
@neeleym I'm just getting around to checking this PR. I'm wondering which parts are still useful. I think our current plan is to comment out and leave much of the old code in for reference for those working on it now. Since you've removed large sections and replaced with TODOs, that may not be as helpful for this new effort. Any thoughts?

**Pull request closed without merging**